Quickly try speeding up CI workflow by parallelizing tasks.

KtorZ opened the pull request:
(work in progress, and very much experimental)

KtorZ committed:
Quickly try speeding up CI workflow by parallelizing tasks.

KtorZ committed:
Trim down nix-shell when running from github workflow

  Avoid hoogle, and unnecessary dev tools.

whs-dot-hk commented on .github/workflows/ci.yaml:
```diff
@@ -8,9 +8,23 @@ on:
+          - package: plutus-merkle-tree
+          - package: hydra-tui
+          - package: hydra-node
+            bench:
```
I see they are not referenced in `steps` as in example https://docs.github.com/en/github-ae@latest/actions/using-jobs/using-a-build-matrix-for-your-jobs#example-running-multiple-versions-of-nodejs only `package` is used

KtorZ replied:
This was simply to check that the workflow syntax was okay :blush: ... Going step-by-step, I wanted to introduce jobs taking care of the benchmarks later. 

Thus indeed, for now, they aren't used.

KtorZ committed:
Add package's name to cache's key.

KtorZ committed:
Trim Nix shell even more.

KtorZ committed:
Add benchmarks to workflow.

KtorZ committed:
Aggregate test results + publish results after all jobs executions

KtorZ committed:
Re-introduce documentation workflow.

KtorZ committed:
Retry TUI tests on failure.

  Using the XTerm terminal seems to be rather flaky and sometimes not start for unknown reason.
